### DISCUSSION TITLE
Könnte das für den PowMr MPPT auch funktionieren?

### DISCUSSION THREAD

**Discussion by Erdbeerfranz** [softwarecrash/EPEver2MQTT#48] [Ideas]
Hallo Softwarecrash, super Projekt, dass du da online gestellt hast. So etwas würde ich für meinen PowMr MPPT auch suchen. Dieser hat eine RS485 Schnittstelle und ich denke von der Hardware wäre das 1:1 das selbe. Programmieren kann ich leider nicht, daher wollte ich fragen ob du sowas auch für den PowMr MPPT hinbekommen würdest? Wäre super, wenn du dich meldest, an Hardware hätte ich alles zum Testen hier. Habe auch versuchsweise dieses Projekt 1:1 so probiert aber leider bekomme ich keine Daten ausgegeben.

vielen Dank vorab!!!

**Comment by softwarecrash** [comment]
Für 1-2 Interessenten sowas zu bauen macht keinen sinn, PowMr ist nur ein rebrander/seller auch die Geräte werden unter x Namen vertrieben, solang weder der eigentliche Hersteller und eine liste der rebrander zu haben ist, und eine vollständige Register liste macht das keinen sinn.
abgesehen davon habe ich nicht so ein teil und im Blindflug irgendwas entwickeln fällt aus.
auch die spenden geben es nicht her mal eben so ein teil zu kaufen um dafür irgendwas zu basteln.

das es aber ModBus ist kannst du dir zb mit tasmota und ein paar rules recht einfach etwas stricken um das teil auszulesen.

**Comment by Erdbeerfranz** [comment]
Hallo, danke für deine Antwort und sorry für den falschen Post unter dem Beitrag, kenne mich leider in Github nicht so aus….ich schaue, dass ich den löschen kann. Sorry nochmalsZum Thema PowMr mit Tasmota auslesen… genau hier scheitere ich leider kläglich, was das zusammenstricken eines Codes angeht. Habe bestimmt schon 15 Stunden dran gesessen aber bekomme es nicht hin. Ich wäre ja schon zufrieden wenn ich irgendwelche Rohdaten rausbekommen würde aus dem Teil aber ich verzweifle daran. Hardware ist alles vorhanden was man bräuchte, obs D1 mini oder ESP32 genauso die MAX RS485 und eine RS485 to TTL Platine(n). Ich würde sogar behaupten ich stelle mich nicht ganz blöd an, mit einer guten Anleitung habe ich schon so manches nachbauen können, nur eben kann ichs nicht wenn ich von 0 beginne :(2 Leute wären wir übrigens schon, die Interesse an dem Code hätten :) v. A. Mit dem Hintergrund, dass man die Daten dann eventuell in Open DTU einbinden könnte. Ich danke dir schon mal für deine Zeit, Gruß, MartinVon meinem iPhone gesendetAm 27.08.2024 um 09:33 schrieb Rocky ***@***.***>:﻿
Für 1-2 Interessenten sowas zu bauen macht keinen sinn, PowMr ist nur ein rebrander/seller auch die Geräte werden unter x Namen vertrieben, solang weder der eigentliche Hersteller und eine liste der rebrander zu haben ist, und eine vollständige Register liste macht das keinen sinn.
abgesehen davon habe ich nicht so ein teil und im Blindflug irgendwas entwickeln fällt aus.
auch die spenden geben es nicht her mal eben so ein teil zu kaufen um dafür irgendwas zu basteln.
das es aber ModBus ist kannst du dir zb mit tasmota und ein paar rules recht einfach etwas stricken um das teil auszulesen.

**Reply by softwarecrash** [reply]
wie schon gesagt ohne so ein gerät zu haben ist das entwickeln schwierig bis unmöglich.
dazu kommt das weder die epever2mqtt struktur noch die solar2mqtt ein direktes weiterreichen nach openDTU erlauben, das passt von da einfach nicht zusammen, da muss eine art broker ala smarthome oder dergleichen dazwischen.

**Comment by Erdbeerfranz** [comment]
OK, das heißt man müsste die Daten erst z. B. an IO Broker schicken und vo da aus weiterverabeiten?

 

 
 

Gesendet: Dienstag, 27. August 2024 um 17:45 Uhr
Von: "Rocky" ***@***.***>
An: "softwarecrash/EPEver2MQTT" ***@***.***>
Cc: "Erdbeerfranz" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [softwarecrash/EPEver2MQTT] Könnte das für den PowMr MPPT auch funktionieren? (Discussion #48)


 

wie schon gesagt ohne so ein gerät zu haben ist das entwickeln schwierig bis unmöglich.
dazu kommt das weder die epever2mqtt struktur noch die solar2mqtt ein direktes weiterreichen nach openDTU erlauben, das passt von da einfach nicht zusammen, da muss eine art broker ala smarthome oder dergleichen dazwischen.

**Reply by softwarecrash** [reply]
ja

**Comment by Erdbeerfranz** [comment]
wie hoch müsste denn die Spende sein, oder würde es reichen, wenn du so einen PowMr "kostenlos" bei dir liegen hättest?
Noch eine Frage zu dem relativ einfachen Code von Tasmota. Tasmota flashen wäre nicht das Problem und ich denke ich brauche die Tasmota Version mit der "Konsole" ?
Du hattest ja geschrieben, dass es mit einem einfachen Code möglich wäre z. M. irgendwas auszulesen aus dem PowMr. Wäre es denn möglich, einen kurzen kurzen Code zu erhalten, damit ich den dann reinkopieren könnte um zu sehen, ob sich irgendwas tut? Ich kann natürlich auch gerne Bilder vom PowMr senden und dem "Aufbau" falls das auch schon hilft


Danke vorab für deine Zeit

**Reply by softwarecrash** [reply]
erstmal wäre halt zu klären wo das teil hingehört, in epever2mqtt würde es vermutlich eher weniger passen da das system eigentlich ausschlieslich auf epever und multi device ausgelegt ist, wird eher zu solar2mqtt tendieren, wobei das bisher nur rs232 nutzt und die von manu entwickelten platinen da wieder nicht passen. muss mit ihm mal durchkauen was sinn macht.
dann bräuchte ich so ein gerät auf dem tisch, vorrausgesetzt die teile lassen sich auch mit irgendeinem kleinen netzteil statt dicker batterie und solar pannels am leben halten um das ganze auch testen zu können.

wenn du IOBroker hast zb dort gibt es einen adapter modbus, ich hab den selber nie genutzt aber da kann man zb ein interface angeben, müsste man mal ru mtüfteln ob man eine modbus tcp bridge nutzen kann dafür. 

in tasmota muss wohl modbus mit einkompiliert werden, dann hat man die möglichkeit entweder über rules bestimmte adressen abzufragen und die antwort davon über mqtt zu senden, oder man fragt die adressen über mqtt ab und bekommt so die antwort.
hier gibt es für den anfang ein paar informationen
https://www.photovoltaikforum.com/thread/211582-rs485-mit-mqtt-verbindung-powmr/

**Reply by Erdbeerfranz** [reply]
Den Beitrag habe ich durchgelesen und den letzen Beitrag der auf Github verlinkt ist habe ich gleich umgesetzt ( der, der mehrere Wechselrichter abfragen kann per Modbus…) 
Leider bekomme ich auch hier keine Rohdaten ausgegeben auch wenn ich hier mit dem Einstellungen rumspiele. 
was mir aber aufgefallen ist, dass nur eine LED von beiden auf der MAX485 Platine blinkt im Abstand von ca 8-10 Sekunden. Es ist auch egal ob man A und B vertauscht, es blinkt immer nur die eine LED….. :(

**Reply by Erdbeerfranz** [reply]
Also es blinkt immer nur die TX LED...

**Reply by all-solutions** [reply]
Wahrscheinlich nutzt das Projekt andere Register und da die Kiste diese nicht kennt, antwortet der nicht.

Du könntest evtl. das hier ausprobieren. Es ist zwar für Hybride, aber vielleicht nutzt es das gleiche Protokoll. Ist zwar auch für RS232 gedacht, es ist aber auch Modbus.

https://github.com/odya/esphome-powmr-hybrid-inverter/tree/main

**Reply by Erdbeerfranz** [reply]
Puh, hab gerade mal drauf geschaut .... da muss ich mich erst wieder einlesen und reinfuchsen... bis jetzt ist mir nicht klar welche software auf den ESP drauf geflashed werden soll... Es gibt aber eine Ordner mit "Register Apps" für den / die Inverter...  Blöd gefragt, kann man dann nicht das Programm das jetzt auf dem ESP32 drauf ist nutzen und nur die "Abfrageadressen" ändern?
Ich will dich nicht in deinem Urlaub damit belästigen, ich würde sagen wir vertagen das ganze um ein paar Tage / Wochen. 
Aber vielen Dank für deine Unterstützung!!!!

**Comment by all-solutions** [comment]
Was hast Du genau für einen MPPT? Also Typ, etc.
Vielleicht wäre es auch gut, ein paar Fotos vom Gerät und auch deren Anschlüssen hier einzustellen.

**Comment by Erdbeerfranz** [comment]
@softwarecrash 
Vielen Dank für die schnelle Antwort. Ich lese mich mal ein und probier ein paar Sachen aus. Den MPPT betreibe ich nur mit 12V über ein kleines Motorrad Batterie Ladegerät. 

@all-solutions 
Ich habe den HHJ60-PRO von PowMr weiß nicht ob ich hier einen Link senden darf, der hat u. a. eine kleine extra Platine für einen Temperatursensor und eben der RS485 Schnittstelle, die vorgesehen ist damit mehrere von den MPPT´s parallel verschaltet werden können und über die Schnittstelle kommunizieren können. Der kostet Bei Aliexpress um die 55-60€ und könnte eben deswegen für einige sehr interessant sein. 
Gerne mache ich ein paar Fotos und poste diese hier. 

Danke euch beiden für eure Zeit

**Reply by all-solutions** [reply]
Von den Spezifikationen her (RS485 und Multi-Device) würde das zur Epever-Platine passen. Die hat ja jetzt schon parallel zur RJ45-Buchse noch eine Schraubleiste mit A/B für den RS485.

Gibt's dafür eine PC-Software? Wenn ja könnte man die genutzten Register mit aufzeichnen. 

Müsste ich mir mal genauer ansehen, wenn ich zurück aus dem Urlaub bin.

**Reply by Erdbeerfranz** [reply]
Leider ist keine Software dabei :(
Ich wünsche dir noch einen schönen Urlaub, danke dir für die ganze Mühe.

**Comment by Erdbeerfranz** [comment]
Hier ein paar Bilder, falls ihr noch weitere Bilder braucht oder Detailbilder wäre das kein Problem. 
![IMG_4917](https://github.com/user-attachments/assets/213fbca8-d94c-441e-8b05-6ff2a151e8b6)
![IMG_4918](https://github.com/user-attachments/assets/c14c5983-b978-4a87-bbaf-f7ca8fd8de12)
![IMG_4919](https://github.com/user-attachments/assets/fad30679-304a-46a0-88fc-f4535e92dc29)
![IMG_4920](https://github.com/user-attachments/assets/c1d7d2a4-a2e7-4192-826c-12c496413538)
![IMG_4921](https://github.com/user-attachments/assets/7d9d9d4e-db2f-4216-bf7b-bb395b4b3be8)
![IMG_4923](https://github.com/user-attachments/assets/f8b5c520-fd4c-4896-926f-e26f5e9c41a6)
![IMG_4924](https://github.com/user-attachments/assets/21124ddf-e9d7-4f32-bbf0-833448bf483d)
![IMG_4926](https://github.com/user-attachments/assets/df71ef43-501c-4042-85fd-a858ab70bc7d)

**Comment by Erdbeerfranz** [comment]
Hardwaretechnisch hätte ich das hier ins Auge gefasst mit 3,3V Spannungsversorgung von unten per 5V auf 3V Wandler
![IMG_4933](https://github.com/user-attachments/assets/b56aef16-b012-4734-9420-e8936ca5058b)
![IMG_4931](https://github.com/user-attachments/assets/c70e8638-0a43-4919-9364-0d6fa18b4dd9)

**Reply by all-solutions** [reply]
Hui, ESP32...
Das wird dann nicht so einfach. Viele der eingesetzten Bibliotheken in den *2MQTT-Projekten sind nicht kompatibel. Da müsste so einiges umgeschrieben werden.

**Reply by Erdbeerfranz** [reply]
Achso, na dann doch eine andere Hardware;) daran soll es  natürlich nicht scheitern

**Comment by Erdbeerfranz** [comment]
Guten Abend zusammen, ich würde ernst machen und einen PowMr zur Verfügung stellen wenn das Angebot noch steht…. 
Könnte man hier per PN oder so in Kontakt treten, damit man die Adresse austauschen kann?

**Reply by softwarecrash** [reply]
du kannst gern jederzeit  in den discord channel kommen https://discord.gg/Hup3gg4YsN
